py/qstr.c

```diff
@@ -324,18 +360,29 @@ qstr qstr_from_strn(const char *str, size_t len) {
         MP_STATE_VM(qstr_last_used) += n_bytes;
 
         // store the interned strings' data
+        #if MICROPY_QSTR_BYTES_IN_HASH
         size_t hash = qstr_compute_hash((const byte *)str, len);
```

Comment:
Instead of wrapping this in `#if`, and having the signature of `qstr_add()` change depending on the config option, maybe it's simpler to move the hash computation inside `qstr_add()`?  Ie:
```c
STATIC qstr qstr_add(mp_uint_t len, const char *q_ptr) {
    size_t hash = qstr_compute_hash(q_ptr, len);
    ...
```